### PR TITLE
Deploy BMO outside of CAPM3 in metal3-dev-env

### PR DESCRIPTION
**Pull request by jan-est**
CAPM3 plans to remove BMO Deployment in its upcoming v1alpha5 api version. This PR deploys the BMO separately and makes the necessary changes to the pivot scripts. 

PRs [242](https://github.com/metal3-io/cluster-api-provider-metal3/pull/242) and [934](https://github.com/metal3-io/baremetal-operator/pull/934) needs to go in before this PR.

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml [hunk 1]
```diff
@@ -23,19 +23,35 @@
...
     kubernetes.core.k8s:
-      name: capm3-ironic
+      name: "{{ NAMEPREFIX }}-ironic"
       kind: Deployment
       state: absent
       namespace: "{{ IRONIC_NAMESPACE }}"
     when: EPHEMERAL_CLUSTER == "minikube"
 
+  - name: Label BMO CRDs.
+    shell: 'kubectl label --overwrite crds baremetalhosts.metal3.io {{ item }}'
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/move.yml:33]: nit:
```suggestion
    shell: "kubectl label --overwrite crds baremetalhosts.metal3.io {{ item }}"
```

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml [hunk 2]
```diff
@@ -82,6 +98,13 @@
   - name: Reinstate Ironic Configmap
     shell: "mv {{ BMOPATH }}/ironic-deployment/keepalived/ironic_bmo_configmap.env.orig {{ BMOPATH }}/ironic-deployment/keepalived/ironic_bmo_configmap.env"
 
+  - name: Label BMO CRDs in target cluster.
+    shell: 'kubectl --kubeconfig /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml label crds baremetalhosts.metal3.io {{ item }} --overwrite '
+    with_items:
+      - clusterctl.cluster.x-k8s.io=""
+      - cluster.x-k8s.io/provider="metal3"
+    when: CAPM3_VERSION == "v1alpha5"
+
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/move.yml:107]: is this for pivoting back and same question with move label here too?

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml [hunk 3]
```diff
@@ -23,19 +23,35 @@
...
       kind: Deployment
       state: absent
       namespace: "{{ IRONIC_NAMESPACE }}"
     when: EPHEMERAL_CLUSTER == "minikube"
 
+  - name: Label BMO CRDs.
+    shell: 'kubectl label --overwrite crds baremetalhosts.metal3.io {{ item }}'
+    with_items:
+       - clusterctl.cluster.x-k8s.io=""
+       - cluster.x-k8s.io/provider="metal3"
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/move.yml:36]: don't we need move label `clusterctl.cluster.x-k8s.io/move` ?

File: vm-setup/roles/v1aX_integration_test/vars/main.yml [hunk 4]
```diff
@@ -2,7 +2,8 @@
 # Ansible playbook related variables
 HOME: "{{ lookup('env', 'HOME') }}"
 NAMESPACE: "{{ lookup('env', 'NAMESPACE') | default('metal3', true) }}"
-IRONIC_NAMESPACE: "{{ lookup('env', 'IRONIC_NAMESPACE') | default('capm3-system', true) }}"
+IRONIC_NAMESPACE: "{{ lookup('env', 'IRONIC_NAMESPACE') | default('baremetal-operator-system', true) }}"
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/vars/main.yml:5]: what if the test is running on v1alpha4?

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml (hunk 3 above)
Comment (jan-est) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml:36]: No we don't need `clusterctl.cluster.x-k8s.io/move` here at all.

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml (hunk 2 above)
Comment (jan-est) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml:107]: Yes, CRD is labeled for pivoting back.

Comment (fmuyassarov) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml:107]: then would it make sense to instead do it within move_back.yaml?

File: vm-setup/roles/v1aX_integration_test/vars/main.yml (hunk 4 above)
Comment (jan-est) [reply vm-setup/roles/v1aX_integration_test/vars/main.yml:5]: Setting the namespace for different api versions is handled [here](https://github.com/metal3-io/metal3-dev-env/blob/5bac171b55a60e9468e1fd7739b7499bd2fa591f/lib/common.sh#L170).

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml (hunk 3 above)
Comment (fmuyassarov) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml:36]: have you tested that this was enough?

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml (hunk 2 above)
Comment (jan-est) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml]: Yes, I think we could do that in move_back.yaml. But in general label is set for all components that we want clusterctl to handle. 

```
// ClusterctlLabelName is applied to all components managed by clusterctl.
	ClusterctlLabelName = "clusterctl.cluster.x-k8s.io"
```

IMO setting these labels is part of the move and pivot operations. Labels should be set right after the move is done for clusterctl to be able to manage the component. So I would leave it here!

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml (hunk 3 above)
Comment (jan-est) [reply vm-setup/roles/v1aX_integration_test/tasks/move.yml:36]: Yes, multiple times!

File: lib/common.sh [hunk 5]
```diff
@@ -215,7 +223,7 @@ export KIND_NODE_IMAGE_VERSION=${KIND_NODE_IMAGE_VERSION:-"v1.21.1"}
 # Minikube version (if EPHEMERAL_CLUSTER=minikube)
 export MINIKUBE_VERSION=${MINIKUBE_VERSION:-"v1.22.0"}
 
-# Ansible version
+# Ansible version
```
Comment (elfosardo) [inline lib/common.sh:226]: nit: blank space at EOL

File: lib/common.sh [hunk 6]
```diff
@@ -405,14 +413,14 @@ differs(){
 #
 function init_minikube() {
     #If the vm exists, it has already been initialized
-    if [[ "$(sudo virsh list --name --all)" != *"minikube"* ]]; then
-      # Restart libvirtd.service as suggested here
+    if [[ "$(sudo virsh list --name --all)" != *"minikube"* ]]; then
```
Comment (elfosardo) [inline lib/common.sh:416]: nit: ditto and below

File: lib/common.sh [hunk 7]
```diff
@@ -405,14 +413,14 @@ differs(){
...
-      # Restart libvirtd.service as suggested here
+    if [[ "$(sudo virsh list --name --all)" != *"minikube"* ]]; then 
+      # Restart libvirtd.service as suggested here 
       # https://github.com/kubernetes/minikube/issues/3566
       sudo systemctl restart libvirtd.service
       # Even if it fails to start minikube here, lets ignore the error
       # It will be retried again in 03 script
       sudo su -l -c "minikube start --insecure-registry ${REGISTRY}" "$USER" || true
-      sudo su -l -c "minikube stop" "$USER" || true
+      sudo su -l -c "minikube stop" "$USER" || true
```
Comment (elfosardo) [inline lib/common.sh:423]: ditto

**Review by furkatgofurov7 — APPROVED** [review]
/approve

**Comment by fmuyassarov** [comment]
@jan-est can you please make sure to fix comments from @elfosardo , I am just going to lgtm right now not to block the whole community
/lgtm

**Pull request merged**